docs/docsite/rst/contributing/creating_collections.rst

```diff
@@ -0,0 +1,138 @@
+.. _creating_content_collections:
+
+********************
+Creating Collections
+********************
+
+.. contents:: Topics
+
+This topic describes how to build a collection and distribute it through Galaxy.
+
+.. _creating_collections:
+
+Collections
+===========
+
+Collections are a distribution format for Ansible content. They can be used to
+package and distribute roles, modules and plugin types.
+
+.. note::
+    This is a Tech-Preview feature. Future Galaxy releases may introduce breaking changes.
+
+
+Collection Metadata
+===================
+
+Collections require a ``galaxy.yml`` at the root level of the collection. This file contains all of the metadata that Galaxy
+and mazer need in order to package and import a collection.
+
+.. note::
+    mazer will only accept ``.yml`` extensions for galaxy.yml.
+
+::
+
+    collection/
+    ├── README.md
+    ├── galaxy.yml
+    ├── plugins/
+    │   ├── modules/
+    │   │   └── module1.py
+    │   ├── inventory/
+    │   └── .../
+    └── roles/
+        ├── role1/
+        └── role2
+
+
+``galaxy.yml`` requires the following fields in order to be accepted by galaxy:
+
+.. code-block:: yaml
+
+    namespace: "namespace_name"
+    name: "collection_name"
+    version: "1.0.12"
+    readme: "README.md"
+    authors:
+        - "Author1"
+        - "Author2 (http://author2.example.com)"
+        - "Author3 <author3@example.com>"
+    dependencies:
+        - "collection.name1"
+        - "collection.name2"
+    description: "Example collection metadata"
+    license:
+        - "MIT"
+    repository: "https://www.github.com/my_org/my_collection"
+    documentation: "http://my-docs.example.com"
+    homepage: "http://www.example.com"
+    issues: "https://www.github.com/my_org/my_collection/issues"
+
+
+Required Fields:
+    - ``namespace`` and ``name`` can only contain alphanumeric characters and underscores.
+      Additionally neither can start with underscores or numbers and cannot contain consecutive
+      underscores.
+    - ``version`` numbers must be compatibly with semantic versioning.
+    - ``readme`` contains the filename for the readme file, which can either be markdown (.md) or
+      reStructuredText (.rst).
+
+
+Optional Fields:
+    - ``dependencies``: A list of collections that this collection depends on.
```

Comment:
Is it worth noting that dependencies cannot be existing repositories on Galaxy? It feels like that would be a really easy thing for someone to assume when they see dependencies like `namespace.thing` which is the syntax they're used to referring to repositories for years.

Reply:
+1 If there is a doc explaining the higher level stuff like 'collections can only depend on other collections (and not roles)' this would be a good place to link to that. That would be a good place to explain that collection dependencies are **install time only** and do not effect runtime ordering, as opposed to galaxy 'roles' that can have install time and/or runtime dependencies.

Reply:
+1 to the above comments from @ironfroggy and @alikins